### DISCUSSION TITLE
AMOUNT FROM URL STRING

### DISCUSSION THREAD

**Discussion by L1mboL** [impress-org/givewp#7319] [Developer Q&A]
Hello.

Can you help me?

I used this code before to fill in amount values from a URL.
It worked with the classic form, but it doesn't work with the new form created in the form builder.

` 
add_action( 'give_post_form_output', 'give_populate_amount', 10, 2 );

function give_populate_amount($form_id, $args) {
	?>

	<script>
			// Get variable from query string in URL
			function giveGetQueryVariable(variable) {
				var query = window.location.search.substring(1);
				var vars = query.split('&');
				for (var i=0;i<vars.length;i++) {
					var pair = vars[i].split('=');
					if(pair[0] == variable){return pair[1];}
				}
				return(false);
			}

			jQuery(document).ready(function( $ ) {

				// Get the amount from the URL
				var getamount = giveGetQueryVariable('amount');
				var amount = '1.00';

				// Set fallback in case URL variable isn't set
				if ( getamount !== false ) {
					amount = getamount;
				}

				// Populate the amount field, then update the total
				
				if ( $('#amount-custom').length > 0 ) {
					$('#amount-custom')
						.val(amount)
						.focus()
						.trigger('blur'} 
				
			

			});

	</script>

**Comment by kjohnson** [comment]
I would suggest using the new Fields API for dynamically modifying fields.

Here is an example using a hidden field, but any field can be used and referenced by name.

See https://givewp.com/documentation/developers/how-to-create-custom-form-fields/#example-four-a-hidden-field-populated-from-a-url-parameter.

**Reply by L1mboL** [reply]
Thank you very much for your response. I have used the method described in the documentation, but the problem remains the same: it works for the classic form and does not work for the form created through the new form builder.
```

add_action('give_post_form_output', function() { ?>
	<script>
		let searchParams = new URLSearchParams(window.location.search);
		// Change the parameter name here
		let amount = searchParams.has('amount') ? searchParams.get('amount') : '';
		// Change the target variable here (amount-custom in new form and give-amount in classic form)
		jQuery("input[name='give-amount']").prop("value", amount);
	</script>
<?php } );

```
